lib/msal-browser/src/app/ClientApplication.ts

```diff
@@ -288,15 +325,30 @@ export abstract class ClientApplication {
         this.logger.verbose("ssoSilent called", request.correlationId);
         this.eventHandler.emitEvent(EventType.SSO_SILENT_START, InteractionType.Silent, request);
 
-        try {
-            const silentIframeClient = new SilentIframeClient(this.config, this.browserStorage, this.browserCrypto, this.logger, this.eventHandler, this.navigationClient, ApiId.ssoSilent, request.correlationId);
-            const silentTokenResult = await silentIframeClient.acquireToken(request);
-            this.eventHandler.emitEvent(EventType.SSO_SILENT_SUCCESS, InteractionType.Silent, silentTokenResult);
-            return silentTokenResult;
-        } catch (e) {
+        let result: Promise<AuthenticationResult>;
+
+        if (this.config.system.platformSSO && this.wamExtensionProvider) {
+            result = this.acquireTokenNative(request).catch((e: AuthError) => {
+                // If native token acquisition fails for availability reasons fallback to standard flow
+                if (e instanceof WamAuthError && e.isFatal()) {
+                    this.wamExtensionProvider = undefined; // If extension gets uninstalled during session prevent future requests from continuing to attempt 
+                    const silentIframeClient = this.createSilentIframeClient(request.correlationId);
+                    return silentIframeClient.acquireToken(request);
+                }
+                throw e;
+            });
+        } else {
+            const silentIframeClient = this.createSilentIframeClient(request.correlationId);
+            result = silentIframeClient.acquireToken(request);
+        }
+
+        return result.then((response) => {
+            this.eventHandler.emitEvent(EventType.SSO_SILENT_SUCCESS, InteractionType.Silent, response);
```

Comment:
Do we want to distinguish WAM API vs Browser API in the event emission?

Reply:
Probably not. These are implementation details, whether the token comes from WAM or elsewhere should be transparent to the user. We can add it later if there's reason to.